### PR TITLE
FoundationDB backend TLS support and housekeeping

### PR DESCRIPTION
**Pull request by jblache**
TLS support was added to FoundationDB in 5.2.4; this PR adds the necessary bits for TLS support to the FoundationDB Vault backend, along with documentation updates.

Some housekeeping is included for good measure. Of note, the FoundationDB Go bindings got vendored in an update, which breaks the build and doesn't work in practice. Revert that and add the proper ignores.

### CODE REVIEW

File: physical/foundationdb/foundationdb.go [hunk 1]
```diff
@@ -174,6 +199,38 @@ func NewFDBBackend(conf map[string]string, logger log.Logger) (physical.Backend,
...
+		if ok {
+			err := opts.SetTLSPassword(tlsPassword)
+			if err != nil {
+				return nil, errwrap.Wrapf("failed to set TLS password: {{err}}", err)
+			}
+		}
+
+		err := opts.SetTLSCaPath(tlsCAFile)
+		if err != nil {
+			return nil, errwrap.Wrapf("failed to set TLS CA bundble path: {{err}}", err)
```
Comment (kalafut) [inline physical/foundationdb/foundationdb.go]:
```suggestion
			return nil, errwrap.Wrapf("failed to set TLS CA bundle path: {{err}}", err)
```

Comment (jblache) [reply physical/foundationdb/foundationdb.go]: Good catch! Fixed :)

File: physical/foundationdb/foundationdb.go [hunk 2]
```diff
@@ -137,6 +140,22 @@ func NewFDBBackend(conf map[string]string, logger log.Logger) (physical.Backend,
 
 	dirPath := strings.Split(strings.Trim(path, "/"), "/")
 
+	// TLS support
+	tlsCertFile, hasCertFile := conf["tls_cert_file"]
+	tlsKeyFile, hasKeyFile := conf["tls_key_file"]
+	tlsCAFile, hasCAFile := conf["tls_ca_file"]
+
+	tlsEnabled := hasCertFile && hasKeyFile && hasCAFile
```
[the remark this answers is not in the document]
Comment (jblache) [reply physical/foundationdb/foundationdb.go]: Thanks! 🎩

File: vendor/vendor.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
 	"comment": "",
-	"ignore": "test",
+	"ignore": "test github.com/apple/foundationdb/bindings/go/src/fdb/",
```
[the remark this answers is not in the document]
Comment (jblache) [reply vendor/vendor.json]: The FoundationDB-related code won't build without installing the bindings first, using the fdb-go-install.sh script. The bindings cannot be vendored for several reasons:
 - they need generated code that isn't checked into git;
 - they need the client library and the versions need to match;
 - the version of the bindings to build with needs to be carefully selected depending on your particular installation, due to features availability and stepped-upgrade requirements.

In short, there is no one-size-fits-all, but, if you follow the backend's documentation, a simple invocation of the fdb-go-install.sh script will get you going for development and test purposes. For production deployments, more FoundationDB knowledge is required, and hopefully I've documented enough of that to at least get folks started on the right path.

In the current state of master, anyone trying to build Vault with FoundationDB will be met with an arcane build failure that they may not be able to get themselves out of :(

**Pull request merged**